### PR TITLE
Add build with docker info to README

### PR DESCRIPTION
**Pull request by alvicsam**
Added information about how to build and test the code with a docker image that is used in our pipelines. (Related to https://github.com/paritytech/ci_cd/issues/113)

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -53,6 +53,22 @@ To run a Rococo collator you will need to compile the following binary:
 cargo build --release -p polkadot-collator
 ```
 
+Otherwise you can compile it with 
+[Parity CI docker image](https://github.com/paritytech/scripts/tree/master/dockerfiles/ci-linux):
+
+```bash
+git clone https://github.com/paritytech/cumulus
```
Comment (chevdor) [inline README.md:60]: I think we can assume that the user cloned the repo and is in the proper folder. The previous commands (`cargo build`...) do assume that.
 
If we **really** want to explain how to clone the repo (I would suggest not to, as it adds maintenance if the repo moves or gets renamed) we could have a section at the beginning.

File: README.md [hunk 2]
```diff
@@ -53,6 +53,22 @@ To run a Rococo collator you will need to compile the following binary:
...
+cd cumulus
+docker pull paritytech/ci-linux:production
```
Comment (chevdor) [inline README.md:62]:
```suggestion
docker pull paritytech/ci-linux:production
```

We don't really need that, the `docker run` will pull as needed.

File: README.md [hunk 3]
```diff
@@ -53,6 +53,22 @@ To run a Rococo collator you will need to compile the following binary:
...
+docker run --rm -it -w /shellhere/cumulus \
+                    -v $(pwd):/shellhere/cumulus \
+                    paritytech/ci-linux:production cargo build --release -p polkadot-collator
```
Comment (chevdor) [inline README.md:65]:
```suggestion
                    paritytech/ci-linux:production cargo build --release --locked -p polkadot-collator
```

**Review by chevdor — CHANGES_REQUESTED** [review]
I think we can make it a little simpler but no big issues.

File: README.md (hunk 3 above)
Comment (chevdor) [reply README.md:62]: yes

**Review by chevdor — APPROVED** [review]
All good once the additionnal missing `--locked` you spoted has been added.

**Review by chevdor — APPROVED** [review]

**Comment by alvicsam** [comment]
@chevdor unfortunately I can't merge changes (the button is inactive), tests are failing. I restarted them several times but with no luck (interesting thing it that each time different tests are failing)

**Comment by chevdor** [comment]
@alvicsam yes I saw that. I just approved **again** after checking that you did the last changes we talked about. 
Don't worry about it the build failing. Your PR has nothing to do with it.
This will be fixed or your PR will be merged "as is". We just need to be patient :)

**Pull request merged**